src/Http/Client/FormDataPart.php

```diff
@@ -219,6 +228,26 @@ public function __toString()
 
         return $out;
     }
+
+    /**
+     * Get the string for the header parameter.
+     *
+     * If the value contains non-ASCII letters an additional header indicating
+     * the charset encoding will be set.
+     *
+     * @param string $name The name of the header parameter
+     * @param string $value The value of the header parameter
+     * @return string
+     */
+    protected function _headerParameterToString($name, $value)
+    {
+        $return = $name . '="' . preg_replace('/[^\x20-\x7E]/', '', $value) . '"';
+        if (!mb_check_encoding($value, 'ASCII') && $this->_charset != null) {
```

Comment:
Same regexp would work. `"` can be added to regexp as it probably not allowed in quotes.